### ISSUE TITLE
Misterious OpenGL error "Invalid operation (1282)"

### ISSUE THREAD

**Issue by riccardobl** [jMonkeyEngine/jmonkeyengine#553]
It seems that when an application that makes use of PBR material is launched with assertion enabled (-ae), at some point,likely when the PBR is rendered, this exception is raised:
```
SEVERE: An OpenGL error has occured!
org.lwjgl.opengl.OpenGLException: Invalid operation (1282)
	at org.lwjgl.opengl.Util.checkGLError(Util.java:59)
	at com.jme3.system.lwjgl.LwjglAbstractDisplay.checkGLError(LwjglAbstractDisplay.java:136)
	at com.jme3.system.lwjgl.LwjglAbstractDisplay.runLoop(LwjglAbstractDisplay.java:157)
	at com.jme3.system.lwjgl.LwjglDisplay.runLoop(LwjglDisplay.java:197)
	at com.jme3.system.lwjgl.LwjglAbstractDisplay.run(LwjglAbstractDisplay.java:232)
	at java.lang.Thread.run(Thread.java:745)
```
However, if assertion are disabled, it seems to run without any noticeable issue.

It has been reported also in this thread: [/t/36292](https://hub.jmonkeyengine.org/t/testpbrlighting-openglexception-invalid-operation-1282/36292).

**Comment by shadowislord** [comment]
@riccardobl 
Please enable graphics debug and try again:
```
AppSettings settings = new AppSettings(true);
settings.putBoolean("GraphicsDebug", true);
app.setSettings(settings);
```

**Comment by riccardobl** [comment]
Code and output here: https://gist.github.com/riccardobl/60673730174b3af3cd1a2b701bb76878

**Comment by shadowislord** [comment]
My running theory is the PBR shader has a mismatch in the interface between the vertex shader and the fragment shader (e.g. varying with different types or non-existent ones).
It still crashes if you remove `FilterPostProcessor` right?

**Comment by Nehon** [comment]
I'll check that... there can be some left overs

**Comment by Nehon** [comment]
I can't reproduce this issue on my end. I've made some cleanup in the j3md, but couldn't find any varying that was not declared in both the vertex and frag shader.
You use the TestPBRLighting, or maybe you have a specific test case for this problem?

**Comment by riccardobl** [comment]
My bad, i was using an outdated build, the issue has been fixed since commit: 85c119c13249139ae561c3aaeeaa38882bfb76f8
From the commit message it looked something mac specific, so i didn't bother updating.

**Comment by Nehon** [comment]
oh ok.
Well an mac I had a black object, so I figured it was mac specific, ut maybe the issue was not as bad on all platform but still there. anyway, glad it's fixed now.
